acra-censor/handlers/whitelist_handler.go

```diff
@@ -29,14 +29,30 @@ func (handler *WhitelistHandler) CheckQuery(query string) (bool, error) {
 		}
 		switch parsedQuery := parsedQuery.(type) {
 		case *sqlparser.Select:
-			err = handler.handleAliasedTables(parsedQuery.From)
-			if err != nil {
-				return false, err
-			}
-			err := handler.handleJoinTables(parsedQuery.From)
-			if err != nil {
-				return false, err
+			for _, fromStatement := range parsedQuery.From {
+				//fmt.Println("Main loop: ", sqlparser.String(fromStatement))
+				switch fromStatement.(type) {
+				case *sqlparser.AliasedTableExpr:
+					err = handler.handleAliasedTables(parsedQuery.From)
+					if err != nil {
+						return false, ErrAccessToForbiddenTableWhitelist
+					}
+					break
+				case *sqlparser.JoinTableExpr:
+					err = handler.handleJoinedTables(fromStatement.(*sqlparser.JoinTableExpr))
+					if err != nil {
+						return false, ErrAccessToForbiddenTableWhitelist
+					}
+				case *sqlparser.ParenTableExpr:
+					err = handler.handleParenTables(fromStatement.(*sqlparser.ParenTableExpr))
+					if err != nil {
+						return false, ErrAccessToForbiddenTableWhitelist
```

Comment:
can you add logging all errors that we wrap with our errors with debug level?
```
log.WithError(err).Debugln("error from WhitlistHandler.handleParentTables")
```
I remember one situation when I had need something like that when debugged censor blocks and need original error to understand - it's parsing error or correct block